internal/pipeline/pipeline.go

```diff
@@ -0,0 +1,55 @@
+package pipeline
+
+import (
+	"fmt"
+	"github.com/mp-hl-2021/lenkeforkortelse/internal/domain/status"
+	"github.com/mp-hl-2021/lenkeforkortelse/internal/usecases/link"
+	"net/http"
+	"time"
+)
+
+const workerCount = 4
+
+func LinkStatusUpdater(luc *link.LinkUseCases) {
+	go func() {
+		links := make(chan link.Link, workerCount)
+
+		go func() {
+			for {
+				userLinks, err := luc.LinkStorage.GetAllUserLinks()
+				if err != nil {
+					fmt.Println("Bad `GetAllUserLinks` request to link database")
+				}
+				for _, lnk := range userLinks {
+					links <- link.Link{LinkId: lnk.LinkId, Link: lnk.Link, LinkStatus: lnk.LinkStatus}
+				}
+				time.Sleep(5 * time.Second)
+			}
+		}()
+
+		for i := 0; i < workerCount; i++ {
+			go func() {
+				for {
+					lnk, ok := <-links
+					if !ok {
+						fmt.Println("links channel has been closed and drained")
+						return
+					}
+					res, _ := http.Get(lnk.Link)
```

Comment:
How shall I put it... 🤔 
So, every network error is just ok? 400 is ok? 500 is ok?

Reply:
Now we treat all status codes greater than or equal to 400 as `Failed`.

Reply:
But you still ignoring http.Get() possible errors...

Reply:
Now we handle http.Get() errors, but we assign `OK` status to any link iff the http.Get() responds successfully or redirects.